hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -498,7 +498,17 @@ case class PatternDef(
     bsym: BlockMemberSymbol,
     tparams: Ls[TyParam],
     paramsOpt: Opt[ParamList],
-    body: ObjBody,
+    /** The pattern parameters, for example, `T` in
+     *  `pattern Nullable(pattern T) = null | T`. */
+    patternParams: Ls[Param],
+    /** The extraction parameters, for example, `x` in
+     *  `pattern PairLike(x, y) = [x, y] | Pair(x, y)`. */
+    extractionParams: Ls[Param],
+    /** The elaborated pattern right-hand side. */
+    pattern: Pattern,
+    // Here, `ObjBody` contains methods `unapply` and `unapplyStringPrefix`,
```

Comment:
I'm not sure this is the right place to store these. Why aren't these methods generated in Lowering? They are implementation details, whereas `Term` should contain user-relevant semantic details.

Reply:
I plan to address this issue in a new PR.